File: src/documentation/CodeSnippets/turtle_snowflake.py
# 4357039f58c42ea585de2bc56af7aa9399fae7086652ebd4b59eddeaa3053112 Elan 1.9.0 guest default_profile valid

def main -> None:  # [main]
  t = new Turtle() # [variable definition]
  t.placeAt(-50, 30) # [call procedure]}
  t.turn(90) # [call procedure]}
  t.show() # [call procedure]}
  for i in range(1, 3, 1):  # [for loop]
    drawSide(side, t) # [call procedure]}
    t.turn(120) # [call procedure]}


def drawSide(length: Float, t: Turtle) -> None:  # [procedure]
  if (length > 1) then
    third = length/3 # [variable definition]
    drawSide(third, t) # [call procedure]}
    t.turn(-60) # [call procedure]}
    drawSide(third, t) # [call procedure]}
    t.turn(120) # [call procedure]}
    drawSide(third, t) # [call procedure]}
    t.turn(-60) # [call procedure]}
    drawSide(third, t) # [call procedure]}
  else
    t.move(length) # [call procedure]}


side = 60

File: test/files/turtle-snowflake.cs
// C# with Elan 2.0.0-beta1

static void main() {
  var t = new Turtle();
  t.placeAt(-50, 30); // procedure call
  t.turn(90); // procedure call
  foreach (var i in range(1, 4)) {
    drawSide(side, t); // procedure call
    t.turn(120); // procedure call
  } // end foreach
} // end main

static void drawSide(double length, Turtle t) { // procedure
  if ((length > 1)) {
    var third = length/3;
    drawSide(third, t); // procedure call
    t.turn(-60); // procedure call
    drawSide(third, t); // procedure call
    t.turn(120); // procedure call
    drawSide(third, t); // procedure call
    t.turn(-60); // procedure call
    drawSide(third, t); // procedure call
  } else {
    t.move(length); // procedure call
  } // end if
} // end procedure

const Int side = 100;
